infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/ReplicatingState.java

```diff
@@ -65,7 +71,10 @@ public LogReplicationRuntimeState processEvent(LogReplicationRuntimeEvent event)
                 fsm.updateConnectedNodes(event.getNodeId());
                 return null;
             case LOCAL_LEADER_LOSS:
-                return fsm.getStates().get(LogReplicationRuntimeStateType.STOPPED);
+                if (canEnqueueStopRuntimeFsmEvent(router, fsm, event.isConnectionStarter)) {
```

Comment:
should we also handle this event in this way in VerifyingRemoteLeaderState?

Reply:
Not needed, because the leadership_loss msg is to be sent only when source is the connection endpoint.
An when SOURCE is the connectionEndpoint, we just pass over the VerifyRemoteLeader state without doing anything as the SINK would be the one sending leadership_query. And the SOURCE, via the query, knows who is the leader on SINK